### ISSUE TITLE
no-lone-blocks doesn't catch lone blocks in SwitchCase statements

### ISSUE THREAD

**Issue by tbassetto** [eslint/eslint#7193]
**What rule do you want to change?**

[`no-lone-blocks`](http://eslint.org/docs/rules/no-lone-blocks)

**Does this change cause the rule to produce more or fewer warnings?**

More warnings hopefully.

**How will the change be implemented? (New option, new default behavior, etc.)?**

A new option to the rule.

**Please provide some example code that this change will affect:**

``` js
var status = 12;
switch (status) {
  case 400:
    {
      console.log('a');
      break;
    }
  default:
    {
      console.log('b');
    }
}
```

**What does the rule currently do for this code?**

It doesn't not complain about the block statements inside the `case` statements. They have no utility here and today there is no way to trigger a warning if they are in the code.

**What will the rule do after it's changed?**

It will trigger a warning.

I first create this issue with the `bug-report.md` template but changed it to the `rule-change-proposal.md` following @platinumazure's advice below.

**Comment by platinumazure** [comment]
It looks like the rule is currently designed to warn for blocks that are created either at the top level (parent is Program node), or which are within another block statement. In this case, the blocks are within a SwitchCase. So this would be an enhancement request to support checking lone blocks under SwitchCase nodes.

We do have a rule `no-case-declarations` which disallow `let`/`const` declarations inside a SwitchCase without being in a nested block, so there is potential for rule conflict. Thus I think an enhancement to `no-lone-blocks` should be behind a new option so people can opt in.

@tbassetto For the reasons above, this is currently working as designed. If you would like to request an enhancement to this rule, could you please edit your first post to reflect [this template](https://github.com/eslint/eslint/blob/master/templates/rule-change-proposal.md)? Thanks!

**Comment by eslintbot** [comment]
Hi @tbassetto, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by tbassetto** [comment]
@platinumazure: done

**Comment by nzakas** [comment]
I think we have to be careful here, as using braces inside `case` statements is a convention I see fairly frequently. I'm on the fence as to whether this makes sense as part of `no-lone-blocks` or if it should be a separate rule, partly because I'm concerned that coming up with an option naming convention that makes clear this particular pattern vs. the ones it currently covers will be tricky.

**Comment by kaicataldo** [comment]
I'm closing this issue because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.
